authentik/lib/avatars.py

```diff
@@ -181,12 +182,41 @@ def avatar_mode_generated(user: "User", mode: str) -> str | None:
 def avatar_mode_url(user: "User", mode: str) -> str | None:
     """Format url"""
     mail_hash = md5(user.email.lower().encode("utf-8"), usedforsecurity=False).hexdigest()  # nosec
-    return mode % {
+
+    formatted_url = mode % {
         "username": user.username,
         "mail_hash": mail_hash,
         "upn": user.attributes.get("upn", ""),
     }
 
+    hostname = urlparse(formatted_url).hostname
+    cache_key_hostname_available = f"goauthentik.io/lib/avatars/{hostname}/available"
+
+    if not cache.get(cache_key_hostname_available, True):
+        return None
+
+    cache_key_full = "goauthentik.io/lib/avatars/{hostname}/avatars/{mail_hash}"
```

Comment:
```suggestion
    cache_key_full = f"goauthentik.io/lib/avatars/{hostname}/{mail_hash}"
```

Feels redundant to repeat "avatars" and fixes a missing fstring

Reply:
thanks, fixed